LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/ConcentratorDeduplication.cs

```diff
@@ -30,96 +39,110 @@ internal sealed class ConcentratorDeduplication<T> :
 
         public ConcentratorDeduplication(
             IMemoryCache cache,
-            WebSocketWriterRegistry<StationEui, string> socketRegistry,
-            ILogger<IConcentratorDeduplication<T>> logger)
+            ILogger<IConcentratorDeduplication> logger)
         {
             this.cache = cache;
-            this.socketRegistry = socketRegistry;
             this.logger = logger;
         }
 
-        public bool ShouldDrop(T frame, StationEui stationEui)
+        public async Task<Result> CheckDuplicateAsync(LoRaRequest loRaRequest, LoRaDevice? loRaDevice)
```

Comment:
I think the logic could be simplified here and made more explicit depending on the request type (join vs. data). If we had 2 explicit entry points, we could avoid special casing the 2 cases in the code. We can still achieve re-usability by having common methods that are called from both. I can help you look into this if you like to see what I have in mind.

Reply:
I can bring back the generics - I decided to remove them in this PR because I got a bit of a pushback on the previous review and especially now that we have a common type (`LoRaRequest`) as an entry point, I decided it was good enough without them. 

Shall we do the change here? Also, if you had another way in mind would be curious to learn :)

Reply:
Changed in 708d47d4e08ada6d83826be47122a5852771349d